Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java

```diff
@@ -215,5 +248,55 @@ public String visualizeAllNetworks(@PathVariable("projectUnixName") String proje
         return "sites/networks/explore";
 
     }
+    
+    /**
+     * This method displays the recent project blog
+     * 
+     * If the project has been set to 'accessible', then the public website page
+     * is displayed. If the project does not exist then an error page is shown.
+     * 
+     * @param unixName
+     *            unix name that is given to the project at the time of its
+     *            creation
+     * @param model
+     *            Model object to map values to view
+     * @return returns a string to access the external website main page
+     * @throws QuadrigaStorageException
+     *             Database storage exception thrown
+     */
+    @RequestMapping(value = "sites/{ProjectUnixName}/projectblogdetails", method = RequestMethod.GET)
+    public String projectblogdetails(Model model, @PathVariable("ProjectUnixName") String unixName, Principal principal,
+            @InjectProject(unixNameParameter = "ProjectUnixName") IProject project) throws QuadrigaStorageException {
+
+        String user = null;
+        if (principal != null) {
+            user = principal.getName();
+        }
+        // Fetch blog entries for a project identified by project unix name
+        String projectId = project.getProjectId();
+        Integer count = 1;
+        List<IProjectBlogEntry> projectBlogEntryList = projectBlogEntryManager.getProjectBlogEntryList(projectId,
```

Comment:
you need to get the blog entry that the user wants to see, not just whatever comes back when you ask for one. In your case, the blog entry that is shown on the home page. But from the blog itself, you also want to link out to details pages and thens how whatever blog entry the user clicked on.